databases/week3/createTables.sql

```diff
@@ -0,0 +1,147 @@
+
+-- Get a meal with any id, fx 1
+SELECT * FROM Meal WHERE id = 1;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Meal SET title = 'biryani', location = 'tivoli'
+WHERE id = 1;
+
+-- Delete a meal with any id, fx 1
+DELETE FROM Meal WHERE id = 3;
+
+
+-- Reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (4, 1, "2019-11-17");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (3, 4, "2019-10-10");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (5, 3, "2019-07-30");
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (2, 2, "2019-9-28");
+
+-- Get all reservations
+SELECT * FROM reservation;
+
+-- Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (8, 9, "2020-08-17");
+
+-- Get a reservation with any id, fx 1
+SELECT * FROM reservation WHERE id = 1;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation SET number_of_guests = 11, meal_id = 8 WHERE id = 1;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation WHERE id = 6;
+
+
+-- Review
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('best dinner', 'I like it very much', 4, 5, "2019-9-28");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Average dinner', 'I like it very much', 3, 5, "2019-11-8");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Excellent dinner', 'I like it very much', 2, 5, "2019-1-28");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Worst dinner', 'I do not like it', 1, 1, "2019-10-2");
+
+
+-- Get all reviews
+SELECT * FROM review;
+
+-- Add a new review
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('nice dinner', 'I like it', 1, 4, "2019-1-2");
+
+-- Get a review with any id, fx 1
+SELECT * FROM review WHERE id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review SET title = 'nice', stars = 3 WHERE id = 1;
+
+-- Delete a review with any id, fx 1
+DELETE FROM review WHERE id = 5;
+
+
+-- Additional queries
+
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM Meal WHERE Price < 4;
+
+-- Get meals that still has available reservations
+SELECT meal.title, meal.max_reservations, reservation.number_of_guests FROM meal JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations >  reservation.number_of_guests;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal WHERE title LIKE '%Rød grød med fløde%'; 
+
+-- Get meals that has been created between two dates
+SELECT * FROM meal
+WHERE created_date BETWEEN '2019-01-5' AND '2030-10-20';
```

Comment:
The exercise does not specify if the date range should be inclusive or exclusive. Using between makes the where clause inclusive. If it should be exclusive then you can also do `WHERE created_date < '2019-01-5' AND created_date > '2030-10-20'`

Reply:
changed WHERE clause to exclusive.